pycbc/workflow/pegasus_workflow.py

```diff
@@ -272,17 +277,15 @@ def root_path(v):
                 path += [v.in_workflow]
```

Comment:
The point of this function is to trace the path that leads to this thing being created. The point being that a dependency can only be placed (and only need to be in any case) between whatever workflow makes the input and the one that the one that consumes it. It doesn't matter if it would be consumed in a subworkflow (except for the fact that pegasus won't know how to get the file, but in terms of dependencies this is correct). 

 I think you need to change this function so that it inludes v as the first element of the ruturn. So initialize as 
```
path = [v]
```
This connects with the following comment.

Reply:
@ahnitz I tried what you suggested here and in the other comment, but it doesn't work. If you change that line to `path = [v]` instead of `path = [v.in_workflow]`, then you never find a `common` workflow between  `self` and `inp`, which causes an error when you try to create the dependency. Here is what I get putting in some print statements:
```
workflow_root: [<pycbc.workflow.core.Workflow object at 0x7f1340b56f60>]
input_root: [<pycbc.workflow.core.Node object at 0x7f1340d8e860>, <pycbc.workflow.core.Workflow object at 0x7f1340ed1d68>]
step: <pycbc.workflow.core.Workflow object at 0x7f1340b56f60>
step in input_root: False
Traceback (most recent call last):
  File "/work/cdcapano/py36-update_pp_workflow/bin/pycbc_make_inference_inj_workflow", line 4, in <module>
    __import__('pkg_resources').run_script('PyCBC===dc0312', 'pycbc_make_inference_inj_workflow')
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/pkg_resources/__init__.py", line 667, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1463, in run_script
    exec(code, namespace, namespace)
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/PyCBC-dc0312-py3.6-linux-x86_64.egg/EGG-INFO/scripts/pycbc_make_inference_inj_workflow", line 364, in <module>
    workflow += pp_workflow
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/PyCBC-dc0312-py3.6-linux-x86_64.egg/pycbc/workflow/pegasus_workflow.py", line 375, in __add__
    return self.add_workflow(other)
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/PyCBC-dc0312-py3.6-linux-x86_64.egg/pycbc/workflow/pegasus_workflow.py", line 317, in add_workflow
    workflow._make_root_dependency(inp.node)
  File "/work/cdcapano/py36-update_pp_workflow/lib/python3.6/site-packages/PyCBC-dc0312-py3.6-linux-x86_64.egg/pycbc/workflow/pegasus_workflow.py", line 291, in _make_root_dependency
    parent=input_root[input_root.index(common)-1].as_job,
UnboundLocalError: local variable 'common' referenced before assignment
```

Reply:
@cdcapano I think you misunderstood. I'm asking you to replace the *initialization* for path with path = [v] i.e. the line that currently says path = []. How it is extended should remain the same.

Reply:
@ahnitz Ah, sorry, I misread your comment. Ok, doing what you suggest worked. Test result is [here](https://www.atlas.aei.uni-hannover.de/~work-cdcapano/scratch/prs/update_pp_workflow/py36_pp_quick_test7_output/html/).